### DISCUSSION TITLE
Relationship model

### DISCUSSION THREAD

**Discussion by empusas** [netbox-community/netbox#8997] [Ideas]
I like NetBox, but I also took a look at Nautobot and found some features that are really useful.

One that struck me was the relationship model that allows to create a relationship models between other object models. For example create a relationship that can be used to tie VLANs to devices, or software versions to devices.
But it is versatile and allows you to create One to One, One to Many, Many to Many relationships between all kind of object models.

I find that this could be a really nice feature that could be used in plugins as well.

**Comment by jeremystretch** [comment]
NetBox v3.2 already includes the ability to create [custom fields referencing one or more other objects](https://netbox.readthedocs.io/en/feature/release-notes/version-3.2/#custom-object-fields-7006), which extends the existing interface for custom fields rather than introducing yet another type of relationship. If you need to model anything more complex than that, you're probably better off creating a custom plugin anyway.

> I find that this could be a really nice feature that could be used in plugins as well.

It would be entirely redundant for a plugin: Plugins can define their own models and relationships using any of the [myriad fields](https://docs.djangoproject.com/en/4.0/ref/models/fields/) provided by the underlying Django framework.

**Reply by empusas** [reply]
I have to admit I did not study the latest updates well. As always good recommendation.
